### DISCUSSION TITLE
OpenXR Support

### DISCUSSION THREAD

**Discussion by 1runeberg** [google/filament#4484] [Ideas]
Re Feature Request: https://github.com/google/filament/issues/4482#issue-965474610

Not quite clear on what you meant here @romainguy since OpenXR deals with more than rendering - is this something on the pipeline?

**Comment by romainguy** [comment]
What I meant is that we don't want to expose rendering backend internals. It would be preferable for us to provide a filament-openxr library or a special backend to connect Filament to OpenXR.

**Reply by 1runeberg** [reply]
thanks for the feedback - so is this something that's already been considered (from a design/architecture perspective)?

i'm still trying to wrap my head around a rendering engine tightly coupled to the openxr api as it handles more than rendering. 

guessing you're thinking solely on swapchain management (perhaps session handling as well) and not the entire openxr sdk?

**Reply by romainguy** [reply]
Yes I'm thinking only about the swap chain management.

**Reply by 1runeberg** [reply]
is this something that's potentially coming up soon? (trying to decide if it's worth creating a fork or just waiting it out)
